### PR TITLE
Add Flow Typings to Status Bar Component

### PR DESCRIPTION
**Pull request by estevanjantsk**
Added Flow typings to status-bar component 👍 .

### CODE REVIEW

**Comment by jdetle** [comment]
@estevanjantsk What happens when you run `npm run flow`? I was having trouble getting it working on my machine this morning.

**Comment by estevanjantsk** [comment]
I tested here with "npm run-script flow" and runned ok. It gives some errors with node_modules, but I think if I put 

```
[ignore]
.*/node_modules/.*
```

in .flowconfig it will ignore all errors =)

File: .flowconfig [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[ignore]
+
+.*node_modules/fbjs.*
```
Comment (captainsafia) [inline .flowconfig]: Can you add the `test` and `static` folders to this as well?

Alternatively, I think you can move this file to `src/` to have it run flow on only that subdirectory.

File: src/notebook/components/status-bar.js [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
+};
+
 export default class StatusBar extends React.Component {
-  static propTypes = {
-    notebook: React.PropTypes.any,
-    lastSaved: React.PropTypes.instanceOf(Date),
-    kernelSpecName: React.PropTypes.string,
-    executionState: React.PropTypes.string,
-  };
+  props: Props;
```
Comment (captainsafia) [inline src/notebook/components/status-bar.js]: Can you move the `props` object into the component like this? The linter gets mad at this right now.

```
props: {
...
}
```

File: package.json [hunk 3]
```diff
@@ -33,7 +33,8 @@
     "dist:linux64": "npm run dist -- --platform=linux --arch=x64",
     "dist:linux32": "npm run dist -- --platform=linux --arch=ia32",
     "dist:osx": "npm run dist -- --platform=darwin --arch=x64 --icon ./static/icon.icns --extend-info=./static/extend.plist --app-category-type=public.app-category.developer-tools --app-bundle-id=io.nteract.nteract",
-    "dist:osx:signed": "npm run dist:osx -- --osx-sign"
+    "dist:osx:signed": "npm run dist:osx -- --osx-sign",
+    "flow": "flow; test $? -eq 0 -o $? -eq 2"
```
Comment (captainsafia) [inline package.json:37]: Do you think you can set up Flow with Babel as well? If not, no worries.

**Review by captainsafia — CHANGES_REQUESTED** [review]

**Comment by captainsafia** [comment]
Thanks for this awesome pull request, Estevan! Welcome to the nteract community! 🎉 

I placed a couple of comments in line for you to take a look at. Let me know if you have any questions about those comments. 😄

**Comment by jdetle** [comment]
_psssst_ You're also welcome to chat on http://slack.nteract.in/

File: src/notebook/components/status-bar.js (hunk 2 above)
Comment (estevanjantsk) [reply src/notebook/components/status-bar.js]: Nice, I thought it would be good practice put with type Props. Im gonna change o/.

Comment (captainsafia) [reply src/notebook/components/status-bar.js]: Yeah, I definitely hear you on that. I think once we've added Flow to all our components we'll make the appropriate updates to our linting setup. Good foresight!

File: package.json (hunk 3 above)
Comment (estevanjantsk) [reply package.json:37]:
```
$> npm install -g babel-cli
$> npm install --save-dev babel-plugin-transform-flow-strip-types
$> echo '{"plugins": ["transform-flow-strip-types"]}' > .babelrc
```

👍

File: .flowconfig (hunk 1 above)
Comment (estevanjantsk) [reply .flowconfig]: ignore these two folders: test and static?

Comment (captainsafia) [reply .flowconfig]: Yes! 👍

**Comment by captainsafia** [comment]
Great! One last thing, we follow the [conventional changelog standard](https://github.com/bcoe/conventional-changelog-standard/blob/master/convention.md) for structuring our commit messages.

Do you know how to rebase your PR and reword your commit messages or would you like us to squash the commits and create a single commit that adheres to the standard?

**Comment by estevanjantsk** [comment]
@captainsafia Sorry but I don't know how to do that =( (rebase your PR and reword your commit messages), may you fix that for me please?

**Comment by jdetle** [comment]
@estevanjantsk, do you use a command line text-editor like vim?

If so, for the future you, you can go in and edit your commits with `git rebase -i HEAD~4`. You use your text editor see each commit in the form of `pick <commit SHA> 'commit message'`. 

From here you look at what you want to change, in this case it might be the commit simply named 'changes'. Rename this by changing 'pick' to 'reword'. After this step, exit the interactive editor. From here you will be taken to a prompt to write a new message. You can update it to what we want from there. When you're satisfied with your changes, you're going to have to 'rewrite history' and use `git push origin <branch_that_you_were_working_on> -f`

Sorry, just wrote this and realized that you two are going in another direction. Awesome PR! Really look forward to learning more about Flow.

**Comment by captainsafia** [comment]
> @captainsafia Sorry but I don't know how to do that =( (rebase your PR and reword your commit messages), may you fix that for me please?

No problem, Estevan! I'll make the appropriate changes on my end. Thank you so much for submitting this pull request!

**Comment by captainsafia** [comment]
@estevanjantsk: Can you give me push access to your fork of this repository? This will allow me to make the changes necessary. To do this you will need to.
1. Go to your fork of the nteract repository.
2. Click "Settings" in the top bar of the repository.
3. Click the "Collaborators" tab and add my username to as a collaborator.

Thanks!

**Review by captainsafia — APPROVED** [review]

**Comment by estevanjantsk** [comment]
@captainsafia added o/!
@jdetle I use atom as my text editor. I know, I kno, I have to go to the dark side and use vim haha. Thanks for the tip with the git.

I'll join the nteract slack, @captainsafia and @jdetle, you both are awesome =DDD!!!

**Comment by jdetle** [comment]
@estevanjantsk I actually use both! `¯\_(ツ)_/¯` I look forward to seeing more PRs in the future!

**Comment by rgbkrk** [comment]
Thanks for kicking off our `flow` setup. :D

**Comment by estevanjantsk** [comment]
@rgbkrk You're welcome o>!

**Pull request merged**